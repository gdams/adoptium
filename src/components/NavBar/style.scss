/* 
---------------------------------------------
Header Style
--------------------------------------------- 
*/

.background-header {
    background-color: #fff;
    height: 80px!important;
    position: fixed!important;
    top: 0!important;
    left: 0;
    right: 0;
    box-shadow: 0 0 10px rgba(0,0,0,15%)!important;
    transition: all .5s ease 0s;
  }
  
  .header-area {
    position: absolute;
    background: #14003C;
    background: linear-gradient(90deg, #14003C 0%, #A80D55 100%);
    height: 110px;
    left: 0;
    right: 0;
    z-index: 100;
    transition: all .5s ease 0s;
  }
  
  .header-area .main-nav {
    min-height: 80px;
    background: transparent;
    display: flex
  }
  
  .header-area .main-nav .logo {
    margin-top: 40px;
    flex-basis: 20%;
    transition: all 0.3s ease 0s;
  }
  
  .background-header .main-nav .logo {
    margin-top: 10px;
  }
  
  .header-area .main-nav .nav {
    flex-basis: 80%;
    height: 110px;
    margin-top: 0;
    margin-right: 0;
    transition: all 0.3s ease 0s;
    position: relative;
    z-index: 999;
  }

  .nav {
    justify-content: right;
  }
  
  .header-area .main-nav .nav li:first-child {
    padding-left: 60px;
  }
  
  .header-area .main-nav .nav li:last-child {
    padding-right: 0;
  }
  
  .header-area .main-nav .nav li {
    padding-left: 30px;
    padding-right: 30px;
  }
  
  .header-area .main-nav .nav li a {
    display: block;
    font-weight: 400;
    font-size: 14px;
    text-transform: uppercase;
    color: #fff;
    transition: all 0.4s ease 0s;
    height: 110px;
    line-height: 110px;
    border: transparent;
    letter-spacing: 1px;
    text-decoration: none;
  }
  
  .background-header .main-nav .nav li a {
    height: 80px;
    line-height: 80px;
  }
  
  .header-area .main-nav .border-button {
    flex-basis: 20%;
    margin-top: 13px;
    text-align: right;
  }
  
  .header-area .main-nav .nav li.has-sub ul.sub-menu li:last-child a:hover,
  .background-header .main-nav .nav li.has-sub ul.sub-menu li:last-child a:hover {
    padding-left: 25px !important;
  }

  .has-sub li,ul {
    padding: 0;
    margin: 0;
    list-style: none;
  }
  
  .header-area .main-nav .nav li:hover a,
  .header-area .main-nav .nav li a.active {
    color: #fff;
    opacity: 0.8;
  }
  
  .background-header .main-nav .nav li:hover a,
  .background-header .main-nav .nav li a.active {
    color: #fff;
    opacity: 0.8;
  }
  
  .header-area .main-nav .nav li.has-sub {
    position: relative;
    padding-right: 15px;
  }

  .background-header .main-nav .nav li.has-sub::after {
    top: 32px;
  }
  
  .header-area .main-nav .nav li.has-sub ul.sub-menu {
    position: absolute;
    width: 250px;
    border-radius: 0 0 5px 5px;
    background-color: #eee;
    box-shadow: 0 2px 28px 0 rgba(0, 0, 0, 6%);
    overflow: hidden;
    top: 110px;
    opacity: 0;
    transition: all .3s;
    transform: translateY(+2em);
    visibility: hidden;
    z-index: -1;
  }
  
  .background-header .main-nav .nav li.has-sub ul.sub-menu {
    top: 80px;
  }
  
  .header-area .main-nav .nav li.has-sub ul.sub-menu li {
    margin-left: 0;
    padding-left: 0;
    padding-right: 0;
  }
  
  .header-area .main-nav .nav li.has-sub ul.sub-menu li a {
    opacity: 1;
    display: block;
    background: #eee;
    color: #2a2a2a!important;
    padding-left: 20px;
    height: 40px;
    line-height: 40px;
    transition: all 0.3s ease 0s;
    position: relative;
    font-size: 13px;
    font-weight: 400;
    border-bottom: 1px solid #eee;
  }
  
  .header-area .main-nav .nav li.has-sub ul.sub-menu li:last-child a {
    border-bottom: none;
  }
  
  .header-area .main-nav .nav li.has-sub ul li a:hover {
    background: #fff;
    color: #FF1464!important;
    padding-left: 25px;
  }
  
  .header-area .main-nav .nav li.has-sub ul li a:hover::before {
    width: 3px;
  }
  
  .header-area .main-nav .nav li.has-sub:hover ul.sub-menu {
    visibility: visible;
    opacity: 1;
    z-index: 1;
    transform: translateY(0%);
    transition-delay: 0s, 0s, 0.3s;
  }
  
  .header-area .main-nav .menu-trigger {
    cursor: pointer;
    position: absolute;
    top: 33px;
    width: 32px;
    height: 40px;
    text-indent: -9999em;
    z-index: 99;
    right: 20px;
    display: none;
  }
  
  .background-header .main-nav .menu-trigger {
    top: 17px;
  }
  
  .header-area .main-nav .menu-trigger span,
  .header-area .main-nav .menu-trigger span::before,
  .header-area .main-nav .menu-trigger span::after {
    transition: all 0.4s;
    background-color: #fff;
    display: block;
    position: absolute;
    width: 30px;
    height: 2px;
    left: 0;
  }
  
  .header-area .main-nav .menu-trigger span::before,
  .header-area .main-nav .menu-trigger span::after {
    transition: all 0.4s;
    background-color: #fff;
    display: block;
    position: absolute;
    height: 2px;
    left: 0;
    width: 75%;
    content: "";
  }
  
  .header-area .main-nav .menu-trigger span {
    top: 16px;
  }
  
  .header-area .main-nav .menu-trigger span::before {
    transform-origin: 33% 100%;
    top: -10px;
    z-index: 10;
  }
  
  .header-area .main-nav .menu-trigger span::after {
    transform-origin: 33% 0;
    top: 10px;
  }
  
  .header-area .main-nav .menu-trigger.active span,
  .header-area .main-nav .menu-trigger.active span::before,
  .header-area .main-nav .menu-trigger.active span::after {
    background-color: transparent;
    width: 100%;
  }
  
  .header-area .main-nav .menu-trigger.active span::before {
    transform: translateY(6px) translateX(1px) rotate(45deg);
    background-color: #fff;
  }
  
  .background-header .main-nav .menu-trigger.active span::before {
    background-color: #fff;
  }
  
  .header-area .main-nav .menu-trigger.active span::after {
    transform: translateY(-6px) translateX(1px) rotate(-45deg);
    background-color: #fff;
  }
  
  .background-header .main-nav .menu-trigger.active span::after {
    background-color: #fff;
  }
  
  .header-area.header-sticky .nav li a.active {
    color: #fff;
  }

  .header-sticky {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    z-index: 999;
  }
  
  @media (width <= 1400px) {
    .header-area .main-nav .nav li {
      padding-left: 10px;
      padding-right: 10px;
    }

    .header-area .main-nav .nav li:first-child {
      padding-left: 0;
    }

    .header-area .main-nav::before {
      display: none;
    }
  }
  
  @media (width <= 1000px) {
    .header-area .main-nav .nav {
      height: auto;
      flex-basis: 100%;
      float: none;
      width: 100%;
      display: none;
      transition: all 0s ease 0s;
      margin-left: -50%;
      max-height: 85vh;
      overflow: auto;
    }

    .header-area .main-nav .logo {
      position: absolute;
      left: 30px;
      top: 0;
    }

    .header-area .main-nav .nav li.has-sub svg {
      position: absolute;
      right: 1em;
      top: 0.5em;
    }

    .background-header .main-nav .logo {
      top: 0;
    }

    .background-header .main-nav .border-button {
      top: 0 !important;
    }

    .header-area .main-nav .border-button {
      position: absolute;
      top: 15px;
      right: 70px;
    }

    .header-area.header-sticky .nav li a:hover,
    .header-area.header-sticky .nav li a.active {
      color: #FF1364!important;
      opacity: 1;
    }

    .header-area.header-sticky .nav li.search-icon a {
      width: 100%;
    }

    .header-area {
      background-color: #f7f7f7;
      padding: 0 15px;
      height: 100px;
      box-shadow: none;
      text-align: center;
    }

    .header-area .container {
      padding: 0;
    }

    .header-area .logo {
      padding-left: 3em;
    }

    .header-area .menu-trigger {
      display: block !important;
    }

    .header-area .main-nav {
      overflow: hidden;
    }

    .header-area .main-nav .nav li:first-child {
      border-top: 1px solid #eee;
    }

    .header-area .nav {
      margin-top: 100px !important;
    }

    .background-header.header-sticky .nav {
      margin-top: 80px !important;
    }

    .header-area .main-nav .nav li {
      width: 100%;
      background: #fff;
      border-bottom: 1px solid #eee;
      padding-left: 0 !important;
      padding-right: 0 !important;
    }

    .header-area .main-nav .nav li a {
      height: 50px !important;
      line-height: 50px !important;
      padding: 0 !important;
      border: none !important;
      background: #f7f7f7 !important;
      color: #191a20 !important;
    }

    .header-area .main-nav .nav li a:hover {
      background: #eee !important;
      color: #FF1364!important;
    }

    .header-area .main-nav .nav li.has-sub ul.sub-menu {
      display: none;
      position: relative;
      visibility: inherit;
      opacity: 1;
      z-index: 1;
      transform: translateY(0%);
      top: 0;
      width: 100%;
      box-shadow: none;
      transition: all 0s;
    }

    .header-area .main-nav .nav li.submenu ul li a {
      font-size: 12px;
      font-weight: 400;
    }

    .header-area .main-nav .nav li.submenu ul li a:hover::before {
      width: 0;
    }

    .header-area .main-nav .nav li.has-sub::after {
      color: #3B566E;
      right: 30px;
      font-size: 14px;
      top: 15px;
    }

    .header-area .main-nav .nav li.submenu:hover ul, .header-area .main-nav .nav li.submenu:focus ul {
      height: 0;
    }
  }

  .pre-header {
    background-color: #fff;
    padding: 15px 0;
  }
  
  .pre-header ul li {
    display: inline-block;
  }
  
  .pre-header .left-info ul li:last-child {
    margin-right: 0;
    padding-right: 0;
    border-right: none;
  }
  
  .pre-header .left-info ul li {
    margin-right: 15px;
    padding-right: 15px;
    border-right: 1px solid #dfb6ff;
  }
  
  .pre-header .left-info ul li a {
    font-size: 14px;
  }
  
  .pre-header .left-info ul li a i {
    font-size: 18px;
    margin-right: 10px;
  }
  
  .pre-header ul li a {
    color: #a586bc;
    transition: all .4s;
  }
  
  .pre-header .social-icons {
    text-align: right;
  }
  
  .pre-header .social-icons ul li {
    margin-left: 10px;
  }
  
  .pre-header .social-icons ul li a {
    font-size: 18px;
  }
  
  .pre-header ul li a:hover {
    color: #c03afe;
  }

  .active {
    display: block !important;
  }
