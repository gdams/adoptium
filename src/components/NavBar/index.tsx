import React, { useState, useEffect } from "react"
import PropTypes from "prop-types"
import { Link } from 'gatsby-plugin-react-i18next';
import { RiArrowDropDownLine, RiArrowDropUpLine } from "react-icons/ri";

import './style.scss';

// @ts-ignore
import Logo from '../../images/adoptium-logo-dark.svg';

const isActive = ({ isCurrent }) => {
  return isCurrent && { className: "active" }
}

const ExactNavLink = props => (
  <Link getProps={isActive}{...props} />
)

const NavBar = (): JSX.Element => {
  const [isSticky, setIsSticky] = useState(false);
  const [isMenuOpen, setIsMenuOpen] = useState(false);
  const [activeSubMenu, setActiveSubMenu] = useState<string | null>(null);

  useEffect(() => {
    const handleScroll = () => {
      if (window.scrollY > 0) {
        setIsSticky(true);
      } else {
        setIsSticky(false);
      }
    }
  
    window.addEventListener('scroll', handleScroll);

    return () => {
      window.removeEventListener('scroll', handleScroll);
    }
  }, []);

  const handleSubMenuClick = (name: string) => {
    // only make changes when in isMenuOpen is true
    if (isMenuOpen) {
      setActiveSubMenu(prevState => prevState === name ? null : name);
    }
  };

  return (
    <header className={`header-area ${isSticky ? 'header-sticky' : ''}`}>
    <div className="container">
      <div className="row">
        <div className="col-12">
          <nav className="main-nav">
            <Link to="/" className="navbar-brand" style={{ marginTop: '2rem' }} aria-label="Homepage Link">
              <Logo alt="Adoptium Logo" style={{ height: '2.5em' }} />
            </Link>
              <ul className={`nav ${isMenuOpen ? 'active' : ''}`}>
              <li>
                <ExactNavLink
                  to="/"
                >
                  Home
                </ExactNavLink>
              </li>
              <li>
                <ExactNavLink
                  to="/marketplace"
                >
                  Marketplace
                </ExactNavLink>
              </li>
              <li>
                <ExactNavLink
                  to="/docs"
                >
                  Documentation
                </ExactNavLink>
              </li>
              <li>
                <ExactNavLink
                  to="/docs/faq"
                >
                  FAQ
                </ExactNavLink>
              </li>
              <li className="has-sub" onClick={() => handleSubMenuClick("Projects")}>
                <a>Projects{activeSubMenu === 'Projects' ? <RiArrowDropUpLine size={30}/> : <RiArrowDropDownLine size={30} />}</a>
                <ul className={`sub-menu ${activeSubMenu === 'Projects' ? 'active' : ''}`}>
                  <li><ExactNavLink className="dropdown-item" to="/temurin">Eclipse Temurin</ExactNavLink></li>
                  <li><ExactNavLink className="dropdown-item" to="/aqavit">Eclipse AQAvit</ExactNavLink></li>
                  <li><ExactNavLink className="dropdown-item" to="/jmc">Eclipse Mission Control</ExactNavLink></li>
                </ul>
              </li>
              <li className="has-sub" onClick={() => handleSubMenuClick("Further Information")}>
                <a>Further Information{activeSubMenu === 'Further Information' ? <RiArrowDropUpLine size={30}/> : <RiArrowDropDownLine size={30} />}</a>
                <ul className={`sub-menu ${activeSubMenu === 'Further Information' ? 'active' : ''}`}>
                  <li><ExactNavLink className="dropdown-item" to="/support">Support</ExactNavLink></li>
                  <li><ExactNavLink className="dropdown-item" to="/news">News & Events</ExactNavLink></li>
                  <li><ExactNavLink className="dropdown-item" to="/about">About</ExactNavLink></li>
                  <li><ExactNavLink className="dropdown-item" to="/members">Members</ExactNavLink></li>
                  <li><ExactNavLink className="dropdown-item" to="/sponsors">Sponsors</ExactNavLink></li>
                  <li><ExactNavLink className="dropdown-item" to="/temurin/buttons">Promote</ExactNavLink></li>
                  <li><a className="dropdown-item" href="https://api.adoptium.net">API</a></li>
                  <li><ExactNavLink className="dropdown-item" to="/blog">Blog</ExactNavLink></li>
                  <li><ExactNavLink className="dropdown-item" to="/support-us">Support Us</ExactNavLink></li>
                  <li><a className="dropdown-item" href="https://status.adoptium.net">Status</a></li>
                </ul>
              </li>
            </ul>
            <a
              className={`menu-trigger ${isMenuOpen ? 'active' : ''}`}
              onClick={() => setIsMenuOpen(!isMenuOpen)}
              data-testid="menu-trigger"
            >
              <span>Menu</span>
            </a>
          </nav>
        </div>
      </div>
    </div>
  </header>
  )
}

NavBar.propTypes = {
  siteTitle: PropTypes.string,
}

NavBar.defaultProps = {
  siteTitle: ``,
}

export default NavBar
