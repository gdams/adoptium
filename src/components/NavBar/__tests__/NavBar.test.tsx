import React from 'react';
import { act, render, fireEvent, screen } from '@testing-library/react';
import { describe, it, expect } from "vitest"
import NavBar from '..';

describe('NavBar component', () => {
  it('navbar renders correctly', () => {
    const { container } = render(
      <NavBar/>
    );
    expect(container).toMatchSnapshot();
  });

  // navbar renders in mobile mode
  it('navbar renders in mobile mode', () => {
    window.innerWidth = 500;
    const { container } = render(
      <NavBar/>
    );
    // check if the menu-trigger is visible
    expect(container.querySelector('.menu-trigger')).toBeInTheDocument();

    // check if the nav is hidden
    expect(container.querySelector('.nav')).not.toHaveClass('active');

    // click the menu-trigger to open the menu
    const menuTrigger = screen.getByTestId('menu-trigger');
    fireEvent.click(menuTrigger);
    expect(container.querySelector('.nav')).toHaveClass('active');
    expect(container.querySelector('.nav .sub-menu')).not.toHaveClass('active');

    // open the Projects sub-menu
    const projectsLink = screen.getByText('Projects');
    expect(projectsLink).toBeInTheDocument();
    fireEvent.click(projectsLink);
  
    expect(container.querySelector('.nav .sub-menu')).toHaveClass('active');

    // click the menu-trigger to close the menu
    fireEvent.click(menuTrigger);
  });

  it('should set isSticky to true when the user scrolls down the page', () => {
    const { container } = render(<NavBar />);
    const header = container.querySelector('header');

    expect(header).toBeInTheDocument();
    expect(header).not.toHaveClass('header-sticky');

    act(() => {
      window.scrollY = 100;
      window.dispatchEvent(new Event('scroll'));
    });

    expect(header).toHaveClass('header-sticky');

    act(() => {
      window.scrollY = 0;
      window.dispatchEvent(new Event('scroll'));
    });

    expect(header).not.toHaveClass('header-sticky');
  });
});
