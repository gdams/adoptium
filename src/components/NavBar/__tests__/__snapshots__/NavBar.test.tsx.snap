// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NavBar component > navbar renders correctly 1`] = `
<div>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-purple"
    style="height: 7rem; padding-top: 1.25em; padding-bottom: 1.25em; position: relative; z-index: 10000;"
  >
    <div
      class="container-fluid"
    >
      <a
        aria-label="Homepage Link"
        class="navbar-brand ms-5"
        href="/"
      >
        <img
          alt="Adoptium Logo"
          style="padding-left: .2em; padding-right: .2em; height: 1.9em;"
        />
      </a>
      <button
        aria-controls="main-navbar"
        aria-expanded="false"
        aria-label="Toggle navigation"
        class="navbar-toggler"
        data-bs-target="#main-navbar"
        data-bs-toggle="collapse"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="collapse navbar-collapse"
        id="main-navbar"
      >
        <ul
          class="navbar-nav bg-purple me-auto mb-2 mb-md-0"
        >
          <li
            class="nav-item"
          >
            <a
              href="/"
            >
              Text
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/marketplace"
            >
              Text
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/docs"
            >
              Text
            </a>
          </li>
          <li
            class="nav-item"
          >
            <a
              href="/docs/faq"
            >
              Text
            </a>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="nav-link dropdown-toggle"
              data-bs-toggle="dropdown"
              href="#"
              id="navbarDropdown"
              role="button"
            >
              Text
            </a>
            <ul
              aria-labelledby="navbarDropdown"
              class="dropdown-menu bg-primary"
            >
              <li>
                <a
                  class="dropdown-item"
                  href="/temurin"
                >
                  Eclipse Temurin
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/aqavit"
                >
                  Eclipse AQAvit
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/jmc"
                >
                  Eclipse Mission Control
                </a>
              </li>
            </ul>
          </li>
          <li
            class="nav-item dropdown"
          >
            <a
              aria-expanded="false"
              class="nav-link dropdown-toggle"
              data-bs-toggle="dropdown"
              href="#"
              id="navbarDropdown2"
              role="button"
            >
              Text
            </a>
            <ul
              aria-labelledby="navbarDropdown2"
              class="dropdown-menu bg-primary"
            >
              <li>
                <a
                  class="dropdown-item"
                  href="/support"
                >
                  Text
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/news"
                >
                  Text
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/about"
                >
                  Text
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/members"
                >
                  Text
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/sponsors"
                >
                  Text
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/adopters"
                >
                  Text
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/temurin/buttons"
                >
                  Text
                </a>
              </li>
              <li>
                <a
                  class="nav-link"
                  href="https://api.adoptium.net"
                >
                  Text
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/blog"
                >
                  Text
                </a>
              </li>
              <li>
                <a
                  class="dropdown-item"
                  href="/support-us"
                >
                  Text
                </a>
              </li>
              <li>
                <a
                  class="nav-link"
                  href="https://status.adoptium.net"
                >
                  Text
                </a>
              </li>
            </ul>
          </li>
        </ul>
      </div>
      <ul
        class="nav col-md-5 col-9 pb-4 justify-content-end list-unstyled d-flex hide-on-mobile p-3"
      >
        <li
          class="ms-3"
        >
          <a
            aria-label="Adoptium Twitter Account"
            href="https://x.com/adoptium"
            rel="noopener noreferrer"
            style="color: rgba(255, 255, 255, 0.55);"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M389.2 48h70.6L305.6 224.2 487 464H345L233.7 318.6 106.5 464H35.8L200.7 275.5 26.8 48H172.4L272.9 180.9 389.2 48zM364.4 421.8h39.1L151.1 88h-42L364.4 421.8z"
              />
            </svg>
          </a>
        </li>
        <li
          class="ms-3"
        >
          <a
            aria-label="Adoptium LinkedIn Page"
            href="https://www.linkedin.com/showcase/adoptium/"
            rel="noopener noreferrer"
            style="color: rgba(255, 255, 255, 0.55);"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
              />
            </svg>
          </a>
        </li>
        <li
          class="ms-3"
        >
          <a
            aria-label="Adoptium YouTube Account"
            href="https://www.youtube.com/c/EclipseAdoptium"
            rel="noopener noreferrer"
            style="color: rgba(255, 255, 255, 0.55);"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 576 512"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M549.655 124.083c-6.281-23.65-24.787-42.276-48.284-48.597C458.781 64 288 64 288 64S117.22 64 74.629 75.486c-23.497 6.322-42.003 24.947-48.284 48.597-11.412 42.867-11.412 132.305-11.412 132.305s0 89.438 11.412 132.305c6.281 23.65 24.787 41.5 48.284 47.821C117.22 448 288 448 288 448s170.78 0 213.371-11.486c23.497-6.321 42.003-24.171 48.284-47.821 11.412-42.867 11.412-132.305 11.412-132.305s0-89.438-11.412-132.305zm-317.51 213.508V175.185l142.739 81.205-142.739 81.201z"
              />
            </svg>
          </a>
        </li>
        <li
          class="ms-3"
        >
          <a
            aria-label="Adoptium GitHub Account"
            href="https://github.com/adoptium"
            rel="noopener noreferrer"
            style="color: rgba(255, 255, 255, 0.55);"
            target="_blank"
          >
            <svg
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 496 512"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              />
            </svg>
          </a>
        </li>
        <li
          class="ms-3"
        >
          <a
            aria-label="Adoptium Slack Account"
            href="/slack"
            style="color: rgba(255, 255, 255, 0.55);"
          >
            <svg
              fill="currentColor"
              height="25"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="25"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M94.12 315.1c0 25.9-21.16 47.06-47.06 47.06S0 341 0 315.1c0-25.9 21.16-47.06 47.06-47.06h47.06v47.06zm23.72 0c0-25.9 21.16-47.06 47.06-47.06s47.06 21.16 47.06 47.06v117.84c0 25.9-21.16 47.06-47.06 47.06s-47.06-21.16-47.06-47.06V315.1zm47.06-188.98c-25.9 0-47.06-21.16-47.06-47.06S139 32 164.9 32s47.06 21.16 47.06 47.06v47.06H164.9zm0 23.72c25.9 0 47.06 21.16 47.06 47.06s-21.16 47.06-47.06 47.06H47.06C21.16 243.96 0 222.8 0 196.9s21.16-47.06 47.06-47.06H164.9zm188.98 47.06c0-25.9 21.16-47.06 47.06-47.06 25.9 0 47.06 21.16 47.06 47.06s-21.16 47.06-47.06 47.06h-47.06V196.9zm-23.72 0c0 25.9-21.16 47.06-47.06 47.06-25.9 0-47.06-21.16-47.06-47.06V79.06c0-25.9 21.16-47.06 47.06-47.06 25.9 0 47.06 21.16 47.06 47.06V196.9zM283.1 385.88c25.9 0 47.06 21.16 47.06 47.06 0 25.9-21.16 47.06-47.06 47.06-25.9 0-47.06-21.16-47.06-47.06v-47.06h47.06zm0-23.72c-25.9 0-47.06-21.16-47.06-47.06 0-25.9 21.16-47.06 47.06-47.06h117.84c25.9 0 47.06 21.16 47.06 47.06 0 25.9-21.16 47.06-47.06 47.06H283.1z"
              />
            </svg>
          </a>
        </li>
      </ul>
    </div>
  </nav>
</div>
`;
